portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -36,30 +38,46 @@
 public class DataServlet extends HttpServlet {
   private final String NAME = "name";
   private final String GET_LOAD_COMMENT_QUANTITY = "loadcommentquantity";
+  private final String USER_EMAIL = "email"; 
 
-  public static String COMMENT = "comment";
+  public static final String COMMENT = "comment";
   
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+
+    //server side Get check: login required to load comments
+    if (!userService.isUserLoggedIn()) {
+      response.sendRedirect("/login_out");
+      return;
+    }
+
+    //user specified comments load quantity should be a number
     int maxCommentsLoad = getUserInput(request);
     if (maxCommentsLoad == -1) {
       response.setContentType("text/html");
       response.getWriter().println("Please enter an integer.");
       return;
     }
 
-    Query query = new Query(COMMENT).addSort("timestamp", SortDirection.DESCENDING);
+    String userEmail = userService.getCurrentUser().getEmail();
 
+    /**search for comments made with the current user's email address
+      start with last comment made
+    */
+    Query query = new Query(COMMENT).setFilter(new Query.FilterPredicate(USER_EMAIL, Query.FilterOperator.EQUAL, userEmail)).addSort("timestamp", SortDirection.DESCENDING);
+    
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
 
     List<Comment> messages = new ArrayList<>();
     for (Entity entity : results.asIterable(FetchOptions.Builder.withLimit(maxCommentsLoad))) {
-      long id = entity.getKey().getId();
+      long messageId = entity.getKey().getId();
       String comment = (String) entity.getProperty(COMMENT);
+      String uEmail = (String) entity.getProperty(USER_EMAIL);
       long timestamp = (long) entity.getProperty("timestamp");
 
-      Comment message = new Comment(id, comment, timestamp);
+      Comment message = new Comment(messageId, comment,  uEmail, timestamp);
```

Comment:
super nit: you have at least 1 extra space in front of uEmail

Reply:
done